Map/MapManager.gd

```diff
@@ -4,18 +4,14 @@ extends Node2D
 static func create_map(): ## Generates and Populates a map with rooms that have random room types. More in depth algorithms will be added in the future
 	var _map: MapBase = MapBase.new()
 	var _grid: Array = [] ## 2d array to return. this will be populated with rooms
-	
-	for i: int in _map.width.size(): ## construct 2d array to return
+	for index_width: int in range(_map.floors_width.size()): ## loop through elements of the grid and assign a room type
 		_grid.append([])
-		for j: int in _map.width[i]:
-			_grid[i].append(0)
-
-	for x: int in range(_map.width.size()): ## loop through elements of the grid and assign a room type
-		for y: int in range(_map.width[x]):
+		for index_height: int in range(_map.floors_width[index_width]):
```

Comment:
On this it should be: `for index_width: int in range(_map.floors_width[index_height]):`
this is logical since you are on the array that gives different width, so an element of it is a width and not a height